### PR TITLE
feat: Added prefix parameter to TiledComponent.load to specify assets folder for tiled maps

### LINKED ISSUE #2641: It would be nice to allow flame_tiled to load assets from directories OTHER than the hardcoded one
# What could be improved

At present, all tiled files must be located in `assets/tiles/`. It would be nice if we could specify an asset path.

# Why should this be improved

This is a problem if the software that implements flame_tiled is being used as a package that another codebase relies on, as we don't have access to `assets/tiles` at that point. Also, if we wanted to download a tsx from the internet and parse that, I am not sure we can write files to the assets directory.

# Any risks?

I suspect that if you follow the same practices as other flame subpackages (ie: `assets/tiles/` is default but changeble) then there would be no issues. I would do it myself but don't know how to write tests for Dart yet.

# More information

I am interested in working on a PR for this, but need to learn how to write Dart tests first.

### PR DESCRIPTION
**Pull request by Guldem**
# Description
Added prefix parameter to `TiledComponent.load` for allowing tiled maps files in different assets folder structures. It still defaults to `assets/tiles`. I created this PR because I discovered it when I wanted to add a tiled map in a different folder structure. I also saw this issue #2641. And though lets add it.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

Closes #2641

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame_tiled/test/tiled_test.dart [hunk 1]
```diff
@@ -40,6 +40,13 @@ void main() {
       expect(tiled.tileMap.renderableLayers.length, equals(3));
     });
 
+    test('correct loads the file, with different prefix', () async {
+      tiled = await TiledComponent.load('map_custom_path.tmx', Vector2.all(16),
+          prefix: 'assets/tiles/tiles_custom_path/');
```
Comment (spydon) [inline packages/flame_tiled/test/tiled_test.dart:45]:
```suggestion
          prefix: 'assets/tiles/tiles_custom_path/',);
```
Just add a trailing comma here and re-format

File: packages/flame_tiled/test/tiled_test.dart [hunk 2]
```diff
@@ -101,6 +108,38 @@ void main() {
...
+        'map.tmx',
+        'tiles_custom_path/external_tileset_custom_path.tsx'
+      ],
+    );
+
+    // TestAssetBundle strips assets/tiles/ from the prefix.
+    final tsxProvider = await FlameTsxProvider.parse(
+        'external_tileset_custom_path.tsx',
+        null,
+        'assets/tiles/tiles_custom_path/');
```
Comment (spydon) [inline packages/flame_tiled/test/tiled_test.dart:126]:
```suggestion
        'assets/tiles/tiles_custom_path/',);
```
Same here

**Review by spydon — APPROVED** [review]
Great, thanks for your contribution!

File: packages/flame_tiled/test/tiled_test.dart (hunk 1 above)
Comment (Guldem) [reply packages/flame_tiled/test/tiled_test.dart]: done

File: packages/flame_tiled/test/tiled_test.dart (hunk 2 above)
Comment (Guldem) [reply packages/flame_tiled/test/tiled_test.dart]: and done

**Pull request merged**